### PR TITLE
ad - Add column called username to leaderboard that maps user id to the correct username

### LINKED ISSUE #25: Approved: Add column called username to leaderboard that maps user id to the correct username

### PR DESCRIPTION
**Pull request by akdas16**
Added a column called 'Username' in the Leaderboard table, and modified its corresponding test to ensure all column headers and content are displayed correctly.

Closes #25

### CODE REVIEW

**Review by singhvyom — APPROVED** [review]
LGTM

**Review by xxzjh — CHANGES_REQUESTED** [review]
I think you should also edit the fixture file for storybook

**Comment by akdas16** [comment]
Do I have to make changes to Leaderboard fixtures (suggested by Jinghan)? I don't think it is necessary.

**Review by pconrad — CHANGES_REQUESTED** [review]
This is not the correct way to do this.

This adds a redundant field to the database, when the username is already associated with the userid.

Instead, we should create a "DTO" (data transfer object) that combines the information from the UserCommons and the User and sends it through a new endpoint, similar to how the CommonsPlus object works.

File: src/main/java/edu/ucsb/cs156/happiercows/entities/UserCommons.java [hunk 1]
```diff
@@ -30,6 +30,10 @@ public class UserCommons {
 
   private int numOfCows;
 
+
+  private String username;
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/happiercows/entities/UserCommons.java:34]: So we did add username to the usercommons in one of the other code bases, but this was a mistake.  We should not have done this, and we are going to have to undo it.

username is already available through the userid, and we should always get it through the user id.  It is bad design to store redundant data in the database.   So this cannot be merged in this form.

Comment (pconrad) [reply src/main/java/edu/ucsb/cs156/happiercows/entities/UserCommons.java:34]: This link shows how this should be implemented: as additional functionality in the UserCommonsController:

https://github.com/ucsb-cs156-f22/f22-7pm-happycows/blob/main/src/main/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsController.java#L145

**Review by pconrad — CHANGES_REQUESTED** [review]
This is not the approach we want to take.  I suggest that we just close this PR without merging.

**Comment by pconrad** [comment]
Closing this without merging.   This is not the approach that we should take to adding username.

The 7pm codebase did this correctly, so we'll use that approach for the S23 codebase.

**Pull request closed without merging**